includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -704,8 +705,8 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 
 			$redirect = $this->get_return_url( $order );
 
-			// If the payment intent requires action, respond with the pi and client secret so it can confirmed on checkout.
-			if ( 'requires_action' === $payment_intent->status ) {
+			// If the payment intent requires action, respond with redirect URL or the pi and client secret so it can confirmed on checkout.
+			if ( 'requires_action' === $payment_intent->status || 'requires_confirmation' === $payment_intent->status ) {
```

Comment:
Hey @james-allan I'm curious if you remember what this change was needed for? Is there a specific gateway that has 'requires_confirmation' as the payment intent status after it has already been confirmed inside `create_and_confirm_payment_intent()`?

I had to do somethign similar to this change in my PR for Boleto and Oxxo but I'm wondering if there's other gateways that need this.

Reply:
When I was working on this PR I needed to change this section so payment methods that need to redirect the user offsite respond to client with the correct redirect URL. Eg this change:  

```php
$redirect = $payment_intent->next_action->redirect_to_url->url;
```

When I was looking into how WooPayments handled this I noticed they had this additional status. I don't recall ever confirming if I received a payment intent status with `requires_confirmation` though. 

You needed it for Boleto and Oxxo though?

Reply:
I noticed you're changes in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2823.

I guess whichever one we merge first, we need to condense what that logic needs to look like.

Reply:
Ive consolidated that logic in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2820/commits/259ab863677dbc56488dabb01c075fff129dbe64